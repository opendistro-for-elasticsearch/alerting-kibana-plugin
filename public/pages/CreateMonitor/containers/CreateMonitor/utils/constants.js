import { OPERATORS_MAP } from '../../../components/MonitorExpressions/expressions/utils/constants';

/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export const BUCKET_COUNT = 5;

export const MATCH_ALL_QUERY = JSON.stringify({ size: 0, query: { match_all: {} } }, null, 4);

export const FORMIK_INITIAL_VALUES = {
  /* CONFIGURE MONITOR */
  name: '',
  disabled: false,
  frequency: 'interval',
  timezone: [],
  daily: 0,
  period: { interval: 1, unit: 'MINUTES' },
  weekly: { mon: false, tue: false, wed: false, thur: false, fri: false, sat: false, sun: false },
  monthly: { type: 'day', day: 1 },
  cronExpression: '0 */1 * * *',

  /* DEFINE MONITOR */
  searchType: 'graph',
  index: [],
  timeField: '',
  query: MATCH_ALL_QUERY,
  aggregationType: 'count',
  fieldName: [],
  overDocuments: 'all documents',
  groupedOverTop: 5,
  groupedOverFieldName: 'bytes',
  bucketValue: 1,
  bucketUnitOfTime: 'h', // m = minute, h = hour, d = day
  where: {
    fieldName: [],
    operator: OPERATORS_MAP.IS,
    fieldValue: '',
    fieldRangeStart: 0,
    fieldRangeEnd: 0,
  },
  detectorId: '',
};
