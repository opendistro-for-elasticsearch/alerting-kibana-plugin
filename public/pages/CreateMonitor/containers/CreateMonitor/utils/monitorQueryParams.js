import { SEARCH_TYPE } from '../../../../../utils/constants';

export const initializeFromQueryParams = queryParams => {
  return {
    searchType: queryParams.searchType || undefined,
    name: queryParams.name ? `${queryParams.name}-Monitor` : undefined,
    detectorId: queryParams.adId || undefined,
    period:
      queryParams.interval && queryParams.unit
        ? { interval: parseInt(queryParams.interval), unit: queryParams.unit }
        : undefined,
  };
};
