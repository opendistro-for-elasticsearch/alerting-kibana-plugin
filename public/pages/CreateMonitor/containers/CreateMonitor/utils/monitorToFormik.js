/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import _ from 'lodash';
import { FORMIK_INITIAL_VALUES } from './constants';
import { SEARCH_TYPE, INPUTS_DETECTOR_ID } from '../../../../../utils/constants';

// Convert Monitor JSON to Formik values used in UI forms
export default function monitorToFormik(monitor) {
  const formikValues = _.cloneDeep(FORMIK_INITIAL_VALUES);
  if (!monitor) return formikValues;
  const {
    name,
    enabled,
    schedule: { cron: { expression: cronExpression = formikValues.cronExpression, timezone } = {} },
    inputs,
    ui_metadata: { schedule = {}, search = {} } = {},
  } = monitor;
  // Default searchType to query, because if there is no ui_metadata or search then it was created through API or overwritten by API
  // In that case we don't want to guess on the UI what selections a user made, so we will default to just showing the extraction query
  const { searchType = 'query', fieldName } = search;
  const isAD = searchType === SEARCH_TYPE.AD;

  return {
    /* INITIALIZE WITH DEFAULTS */
    ...formikValues,

    /* CONFIGURE MONITOR */
    name,
    disabled: !enabled,

    /* This will overwrite the fields in use by Monitor from ui_metadata */
    ...schedule,
    cronExpression,

    /* DEFINE MONITOR */
    ...search,
    searchType,
    fieldName: fieldName ? [{ label: fieldName }] : [],
    timezone: timezone ? [{ label: timezone }] : [],

    detectorId: isAD //&& _.get(inputs, '0.search.indices.0') === '.opendistro-anomaly-results*'
      ? _.get(inputs, INPUTS_DETECTOR_ID)
      : undefined,
    index: inputs[0].search.indices.map(index => ({ label: index })),
    query: JSON.stringify(inputs[0].search.query, null, 4),
  };
}
