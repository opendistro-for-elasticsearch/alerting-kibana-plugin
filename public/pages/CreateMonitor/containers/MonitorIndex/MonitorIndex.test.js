/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import { Formik } from 'formik';
import { mount } from 'enzyme';

import { FORMIK_INITIAL_VALUES } from '../CreateMonitor/utils/constants';
import MonitorIndex from './MonitorIndex';
import * as helpers from './utils/helpers';
import { httpClientMock } from '../../../../../test/mocks';

helpers.createReasonableWait = jest.fn(cb => cb());
httpClientMock.post.mockResolvedValue({ data: { ok: true, resp: [] } });

function getMountWrapper(customProps = {}) {
  return mount(
    <Formik
      initialValues={FORMIK_INITIAL_VALUES}
      render={() => <MonitorIndex httpClient={httpClientMock} {...customProps} />}
    />
  );
}

describe('MonitorIndex', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('renders', () => {
    const wrapper = getMountWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  test('calls onSearchChange when changing input value', async () => {
    const onSearchChange = jest.spyOn(MonitorIndex.prototype, 'onSearchChange');
    const wrapper = getMountWrapper();
    wrapper
      .find('[data-test-subj="comboBoxSearchInput"]')
      .hostNodes()
      .simulate('change', { target: { value: 'random-index' } });
    setTimeout(() => {
      wrapper.update();
      expect(onSearchChange).toHaveBeenCalled();
      expect(onSearchChange).toHaveBeenCalledWith('random-index');
      done();
    });
  });

  test('appends wildcard when search is one valid character', () => {
    const wrapper = getMountWrapper();

    wrapper
      .find('[data-test-subj="comboBoxSearchInput"]')
      .hostNodes()
      .simulate('change', { target: { value: 'r' } });

    expect(wrapper.find(MonitorIndex).instance().state.appendedWildcard).toBe(true);
    expect((wrapper.find(MonitorIndex).instance().lastQuery = 'r*'));
  });

  test('searches space normalizes value', () => {
    const wrapper = getMountWrapper();

    wrapper
      .find('[data-test-subj="comboBoxSearchInput"]')
      .hostNodes()
      .simulate('change', { target: { value: ' ' } })
      .simulate('keyDown', { keyCode: 13 });

    expect(wrapper.find('.euiComboBoxPill')).toHaveLength(0);
  });

  test('searches resets appendedWildcard', () => {
    const wrapper = getMountWrapper();

    wrapper
      .find('[data-test-subj="comboBoxSearchInput"]')
      .hostNodes()
      .simulate('change', { target: { value: 'r' } });

    expect(wrapper.find(MonitorIndex).instance().state.appendedWildcard).toBe(true);
    expect((wrapper.find(MonitorIndex).instance().lastQuery = 'r*'));

    wrapper
      .find('[data-test-subj="comboBoxSearchInput"]')
      .hostNodes()
      .simulate('change', { target: { value: '*' } });

    expect(wrapper.find(MonitorIndex).instance().state.appendedWildcard).toBe(false);
    expect((wrapper.find(MonitorIndex).instance().lastQuery = ''));
  });

  test('returns empty alias/index array for *:', async () => {
    const wrapper = getMountWrapper();

    expect(
      await wrapper
        .find(MonitorIndex)
        .instance()
        .handleQueryAliases('*:')
    ).toEqual([]);
    expect(
      await wrapper
        .find(MonitorIndex)
        .instance()
        .handleQueryIndices('*:')
    ).toEqual([]);
  });

  test('returns empty array for data.ok = false', async () => {
    httpClientMock.post.mockResolvedValue({ data: { ok: false } });
    const wrapper = getMountWrapper();

    expect(
      await wrapper
        .find(MonitorIndex)
        .instance()
        .handleQueryAliases('random')
    ).toEqual([]);
    expect(
      await wrapper
        .find(MonitorIndex)
        .instance()
        .handleQueryIndices('random')
    ).toEqual([]);
  });
  //
  test('returns indices/aliases', async () => {
    httpClientMock.post.mockResolvedValue({
      data: {
        ok: true,
        resp: [{ health: 'green', status: 'open', index: 'logstash-0', alias: 'logstash' }],
      },
    });
    const wrapper = getMountWrapper();

    expect(
      await wrapper
        .find(MonitorIndex)
        .instance()
        .handleQueryAliases('l')
    ).toEqual([{ label: 'logstash', index: 'logstash-0' }]);
    expect(
      await wrapper
        .find(MonitorIndex)
        .instance()
        .handleQueryIndices('l')
    ).toEqual([{ health: 'green', status: 'open', label: 'logstash-0' }]);
  });

  test.skip('onBlur sets index to touched', () => {
    httpClientMock.post.mockResolvedValue({
      data: {
        ok: true,
        resp: [{ health: 'green', status: 'open', index: 'logstash-0', alias: 'logstash' }],
      },
    });
    const wrapper = getMountWrapper();

    wrapper
      .find('[data-test-subj="comboBoxSearchInput"]')
      .hostNodes()
      .simulate('change', { target: { value: 'l' } })
      .simulate('blur');

    expect(wrapper.instance().state.touched).toEqual({ index: true });
  });

  test('sets option when calling onCreateOption', () => {
    httpClientMock.post.mockResolvedValue({
      data: {
        ok: true,
        resp: [{ health: 'green', status: 'open', index: 'logstash-0', alias: 'logstash' }],
      },
    });
    const wrapper = getMountWrapper();

    wrapper
      .find('[data-test-subj="comboBoxSearchInput"]')
      .hostNodes()
      .simulate('change', { target: { value: 'logstash-0' } });

    wrapper
      .find('[data-test-subj="comboBoxInput"]')
      .hostNodes()
      .simulate('keyDown', { keyCode: 40 })
      .simulate('keyDown', { keyCode: 13 });

    expect(wrapper.instance().state.values.index).toEqual([{ label: 'logstash-0' }]);
  });
});
