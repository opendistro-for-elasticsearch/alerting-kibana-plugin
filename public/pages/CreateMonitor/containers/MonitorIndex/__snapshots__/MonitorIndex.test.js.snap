// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitorIndex renders 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "aggregationType": "count",
      "bucketUnitOfTime": "h",
      "bucketValue": 1,
      "cronExpression": "0 */1 * * *",
      "daily": 0,
      "disabled": false,
      "fieldName": Array [],
      "frequency": "interval",
      "groupedOverFieldName": "bytes",
      "groupedOverTop": 5,
      "index": Array [],
      "monthly": Object {
        "day": 1,
        "type": "day",
      },
      "name": "",
      "overDocuments": "all documents",
      "period": Object {
        "interval": 1,
        "unit": "MINUTES",
      },
      "query": "{
    \\"size\\": 0,
    \\"query\\": {
        \\"match_all\\": {}
    }
}",
      "searchType": "graph",
      "timeField": "",
      "timezone": Array [],
      "weekly": Object {
        "fri": false,
        "mon": false,
        "sat": false,
        "sun": false,
        "thur": false,
        "tue": false,
        "wed": false,
      },
      "where": Object {
        "fieldName": Array [],
        "fieldRangeEnd": 0,
        "fieldRangeStart": 0,
        "fieldValue": 0,
        "operator": "is",
      },
    }
  }
  isInitialValid={false}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
>
  <MonitorIndex
    httpClient={[MockFunction]}
  >
    <FormikComboBox
      fieldProps={
        Object {
          "validate": [Function],
        }
      }
      formRow={true}
      inputProps={
        Object {
          "async": true,
          "data-test-subj": "indicesComboBox",
          "isClearable": true,
          "isLoading": true,
          "onBlur": [Function],
          "onChange": [Function],
          "onCreateOption": [Function],
          "onSearchChange": [Function],
          "options": Array [
            Object {
              "label": "Aliases",
              "options": Array [],
            },
            Object {
              "label": "Indices",
              "options": Array [],
            },
          ],
          "placeholder": "Select indices",
          "renderOption": [Function],
        }
      }
      name="index"
      rowProps={
        Object {
          "error": [Function],
          "helpText": "You can use a * as a wildcard in your index pattern",
          "isInvalid": [Function],
          "label": "Index",
          "style": Object {
            "paddingLeft": "10px",
          },
        }
      }
    >
      <FormikInputWrapper
        fieldProps={
          Object {
            "validate": [Function],
          }
        }
        name="index"
        render={[Function]}
      >
        <C
          name="index"
          render={[Function]}
          validate={[Function]}
        >
          <FieldInner
            formik={
              Object {
                "dirty": false,
                "errors": Object {},
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialValues": Object {
                  "aggregationType": "count",
                  "bucketUnitOfTime": "h",
                  "bucketValue": 1,
                  "cronExpression": "0 */1 * * *",
                  "daily": 0,
                  "disabled": false,
                  "fieldName": Array [],
                  "frequency": "interval",
                  "groupedOverFieldName": "bytes",
                  "groupedOverTop": 5,
                  "index": Array [],
                  "monthly": Object {
                    "day": 1,
                    "type": "day",
                  },
                  "name": "",
                  "overDocuments": "all documents",
                  "period": Object {
                    "interval": 1,
                    "unit": "MINUTES",
                  },
                  "query": "{
    \\"size\\": 0,
    \\"query\\": {
        \\"match_all\\": {}
    }
}",
                  "searchType": "graph",
                  "timeField": "",
                  "timezone": Array [],
                  "weekly": Object {
                    "fri": false,
                    "mon": false,
                    "sat": false,
                    "sun": false,
                    "thur": false,
                    "tue": false,
                    "wed": false,
                  },
                  "where": Object {
                    "fieldName": Array [],
                    "fieldRangeEnd": 0,
                    "fieldRangeStart": 0,
                    "fieldValue": 0,
                    "operator": "is",
                  },
                },
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setError": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validate": undefined,
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validationSchema": undefined,
                "values": Object {
                  "aggregationType": "count",
                  "bucketUnitOfTime": "h",
                  "bucketValue": 1,
                  "cronExpression": "0 */1 * * *",
                  "daily": 0,
                  "disabled": false,
                  "fieldName": Array [],
                  "frequency": "interval",
                  "groupedOverFieldName": "bytes",
                  "groupedOverTop": 5,
                  "index": Array [],
                  "monthly": Object {
                    "day": 1,
                    "type": "day",
                  },
                  "name": "",
                  "overDocuments": "all documents",
                  "period": Object {
                    "interval": 1,
                    "unit": "MINUTES",
                  },
                  "query": "{
    \\"size\\": 0,
    \\"query\\": {
        \\"match_all\\": {}
    }
}",
                  "searchType": "graph",
                  "timeField": "",
                  "timezone": Array [],
                  "weekly": Object {
                    "fri": false,
                    "mon": false,
                    "sat": false,
                    "sun": false,
                    "thur": false,
                    "tue": false,
                    "wed": false,
                  },
                  "where": Object {
                    "fieldName": Array [],
                    "fieldRangeEnd": 0,
                    "fieldRangeStart": 0,
                    "fieldValue": 0,
                    "operator": "is",
                  },
                },
              }
            }
            name="index"
            render={[Function]}
            validate={[Function]}
          >
            <FormikFormRow
              form={
                Object {
                  "dirty": false,
                  "errors": Object {},
                  "handleBlur": [Function],
                  "handleChange": [Function],
                  "handleReset": [Function],
                  "handleSubmit": [Function],
                  "initialValues": Object {
                    "aggregationType": "count",
                    "bucketUnitOfTime": "h",
                    "bucketValue": 1,
                    "cronExpression": "0 */1 * * *",
                    "daily": 0,
                    "disabled": false,
                    "fieldName": Array [],
                    "frequency": "interval",
                    "groupedOverFieldName": "bytes",
                    "groupedOverTop": 5,
                    "index": Array [],
                    "monthly": Object {
                      "day": 1,
                      "type": "day",
                    },
                    "name": "",
                    "overDocuments": "all documents",
                    "period": Object {
                      "interval": 1,
                      "unit": "MINUTES",
                    },
                    "query": "{
    \\"size\\": 0,
    \\"query\\": {
        \\"match_all\\": {}
    }
}",
                    "searchType": "graph",
                    "timeField": "",
                    "timezone": Array [],
                    "weekly": Object {
                      "fri": false,
                      "mon": false,
                      "sat": false,
                      "sun": false,
                      "thur": false,
                      "tue": false,
                      "wed": false,
                    },
                    "where": Object {
                      "fieldName": Array [],
                      "fieldRangeEnd": 0,
                      "fieldRangeStart": 0,
                      "fieldValue": 0,
                      "operator": "is",
                    },
                  },
                  "isSubmitting": false,
                  "isValid": false,
                  "isValidating": false,
                  "registerField": [Function],
                  "resetForm": [Function],
                  "setError": [Function],
                  "setErrors": [Function],
                  "setFieldError": [Function],
                  "setFieldTouched": [Function],
                  "setFieldValue": [Function],
                  "setFormikState": [Function],
                  "setStatus": [Function],
                  "setSubmitting": [Function],
                  "setTouched": [Function],
                  "setValues": [Function],
                  "submitCount": 0,
                  "submitForm": [Function],
                  "touched": Object {},
                  "unregisterField": [Function],
                  "validateField": [Function],
                  "validateForm": [Function],
                  "validateOnBlur": true,
                  "validateOnChange": true,
                  "values": Object {
                    "aggregationType": "count",
                    "bucketUnitOfTime": "h",
                    "bucketValue": 1,
                    "cronExpression": "0 */1 * * *",
                    "daily": 0,
                    "disabled": false,
                    "fieldName": Array [],
                    "frequency": "interval",
                    "groupedOverFieldName": "bytes",
                    "groupedOverTop": 5,
                    "index": Array [],
                    "monthly": Object {
                      "day": 1,
                      "type": "day",
                    },
                    "name": "",
                    "overDocuments": "all documents",
                    "period": Object {
                      "interval": 1,
                      "unit": "MINUTES",
                    },
                    "query": "{
    \\"size\\": 0,
    \\"query\\": {
        \\"match_all\\": {}
    }
}",
                    "searchType": "graph",
                    "timeField": "",
                    "timezone": Array [],
                    "weekly": Object {
                      "fri": false,
                      "mon": false,
                      "sat": false,
                      "sun": false,
                      "thur": false,
                      "tue": false,
                      "wed": false,
                    },
                    "where": Object {
                      "fieldName": Array [],
                      "fieldRangeEnd": 0,
                      "fieldRangeStart": 0,
                      "fieldValue": 0,
                      "operator": "is",
                    },
                  },
                }
              }
              name="index"
              rowProps={
                Object {
                  "error": [Function],
                  "helpText": "You can use a * as a wildcard in your index pattern",
                  "isInvalid": [Function],
                  "label": "Index",
                  "style": Object {
                    "paddingLeft": "10px",
                  },
                }
              }
            >
              <EuiFormRow
                describedByIds={Array []}
                fullWidth={false}
                hasEmptyLabelSpace={false}
                helpText="You can use a * as a wildcard in your index pattern"
                id="index"
                isInvalid={false}
                label="Index"
                labelType="label"
                style={
                  Object {
                    "paddingLeft": "10px",
                  }
                }
              >
                <div
                  className="euiFormRow"
                  id="index-row"
                  style={
                    Object {
                      "paddingLeft": "10px",
                    }
                  }
                >
                  <div>
                    <EuiFormLabel
                      aria-invalid={false}
                      htmlFor="index"
                      isFocused={false}
                      isInvalid={false}
                      type="label"
                    >
                      <label
                        aria-invalid={false}
                        className="euiFormLabel"
                        htmlFor="index"
                      >
                        Index
                      </label>
                    </EuiFormLabel>
                  </div>
                  <ComboBox
                    aria-describedby="index-help"
                    field={
                      Object {
                        "name": "index",
                        "onBlur": [Function],
                        "onChange": [Function],
                        "value": Array [],
                      }
                    }
                    form={
                      Object {
                        "dirty": false,
                        "errors": Object {},
                        "handleBlur": [Function],
                        "handleChange": [Function],
                        "handleReset": [Function],
                        "handleSubmit": [Function],
                        "initialValues": Object {
                          "aggregationType": "count",
                          "bucketUnitOfTime": "h",
                          "bucketValue": 1,
                          "cronExpression": "0 */1 * * *",
                          "daily": 0,
                          "disabled": false,
                          "fieldName": Array [],
                          "frequency": "interval",
                          "groupedOverFieldName": "bytes",
                          "groupedOverTop": 5,
                          "index": Array [],
                          "monthly": Object {
                            "day": 1,
                            "type": "day",
                          },
                          "name": "",
                          "overDocuments": "all documents",
                          "period": Object {
                            "interval": 1,
                            "unit": "MINUTES",
                          },
                          "query": "{
    \\"size\\": 0,
    \\"query\\": {
        \\"match_all\\": {}
    }
}",
                          "searchType": "graph",
                          "timeField": "",
                          "timezone": Array [],
                          "weekly": Object {
                            "fri": false,
                            "mon": false,
                            "sat": false,
                            "sun": false,
                            "thur": false,
                            "tue": false,
                            "wed": false,
                          },
                          "where": Object {
                            "fieldName": Array [],
                            "fieldRangeEnd": 0,
                            "fieldRangeStart": 0,
                            "fieldValue": 0,
                            "operator": "is",
                          },
                        },
                        "isSubmitting": false,
                        "isValid": false,
                        "isValidating": false,
                        "registerField": [Function],
                        "resetForm": [Function],
                        "setError": [Function],
                        "setErrors": [Function],
                        "setFieldError": [Function],
                        "setFieldTouched": [Function],
                        "setFieldValue": [Function],
                        "setFormikState": [Function],
                        "setStatus": [Function],
                        "setSubmitting": [Function],
                        "setTouched": [Function],
                        "setValues": [Function],
                        "submitCount": 0,
                        "submitForm": [Function],
                        "touched": Object {},
                        "unregisterField": [Function],
                        "validateField": [Function],
                        "validateForm": [Function],
                        "validateOnBlur": true,
                        "validateOnChange": true,
                        "values": Object {
                          "aggregationType": "count",
                          "bucketUnitOfTime": "h",
                          "bucketValue": 1,
                          "cronExpression": "0 */1 * * *",
                          "daily": 0,
                          "disabled": false,
                          "fieldName": Array [],
                          "frequency": "interval",
                          "groupedOverFieldName": "bytes",
                          "groupedOverTop": 5,
                          "index": Array [],
                          "monthly": Object {
                            "day": 1,
                            "type": "day",
                          },
                          "name": "",
                          "overDocuments": "all documents",
                          "period": Object {
                            "interval": 1,
                            "unit": "MINUTES",
                          },
                          "query": "{
    \\"size\\": 0,
    \\"query\\": {
        \\"match_all\\": {}
    }
}",
                          "searchType": "graph",
                          "timeField": "",
                          "timezone": Array [],
                          "weekly": Object {
                            "fri": false,
                            "mon": false,
                            "sat": false,
                            "sun": false,
                            "thur": false,
                            "tue": false,
                            "wed": false,
                          },
                          "where": Object {
                            "fieldName": Array [],
                            "fieldRangeEnd": 0,
                            "fieldRangeStart": 0,
                            "fieldValue": 0,
                            "operator": "is",
                          },
                        },
                      }
                    }
                    id="index"
                    inputProps={
                      Object {
                        "async": true,
                        "data-test-subj": "indicesComboBox",
                        "isClearable": true,
                        "isLoading": true,
                        "onBlur": [Function],
                        "onChange": [Function],
                        "onCreateOption": [Function],
                        "onSearchChange": [Function],
                        "options": Array [
                          Object {
                            "label": "Aliases",
                            "options": Array [],
                          },
                          Object {
                            "label": "Indices",
                            "options": Array [],
                          },
                        ],
                        "placeholder": "Select indices",
                        "renderOption": [Function],
                      }
                    }
                    name="index"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    <EuiComboBox
                      async={true}
                      compressed={false}
                      data-test-subj="indicesComboBox"
                      fullWidth={false}
                      id="index"
                      isClearable={true}
                      isLoading={true}
                      name="index"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onCreateOption={[Function]}
                      onSearchChange={[Function]}
                      options={
                        Array [
                          Object {
                            "label": "Aliases",
                            "options": Array [],
                          },
                          Object {
                            "label": "Indices",
                            "options": Array [],
                          },
                        ]
                      }
                      placeholder="Select indices"
                      renderOption={[Function]}
                      selectedOptions={Array []}
                      singleSelection={false}
                    >
                      <div
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        className="euiComboBox"
                        data-test-subj="indicesComboBox"
                        name="index"
                        onKeyDown={[Function]}
                        role="combobox"
                      >
                        <EuiComboBoxInput
                          autoSizeInputRef={[Function]}
                          compressed={false}
                          focusedOptionId={null}
                          fullWidth={false}
                          hasSelectedOptions={false}
                          id="index"
                          inputRef={[Function]}
                          isListOpen={false}
                          noIcon={false}
                          onChange={[Function]}
                          onClear={[Function]}
                          onClick={[Function]}
                          onCloseListClick={[Function]}
                          onFocus={[Function]}
                          onOpenListClick={[Function]}
                          onRemoveOption={[Function]}
                          placeholder="Select indices"
                          rootId={[Function]}
                          searchValue=""
                          selectedOptions={Array []}
                          singleSelection={false}
                          toggleButtonRef={[Function]}
                          updatePosition={[Function]}
                          value=""
                        >
                          <EuiFormControlLayout
                            compressed={false}
                            fullWidth={false}
                            icon={
                              Object {
                                "aria-label": "Open list of options",
                                "data-test-subj": "comboBoxToggleListButton",
                                "disabled": undefined,
                                "onClick": [Function],
                                "ref": [Function],
                                "side": "right",
                                "type": "arrowDown",
                              }
                            }
                            isLoading={false}
                          >
                            <div
                              className="euiFormControlLayout"
                            >
                              <div
                                className="euiFormControlLayout__childrenWrapper"
                              >
                                <div
                                  className="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                                  data-test-subj="comboBoxInput"
                                  onClick={[Function]}
                                  tabIndex="-1"
                                >
                                  <p
                                    className="euiComboBoxPlaceholder"
                                  >
                                    Select indices
                                  </p>
                                  <AutosizeInput
                                    aria-activedescendant={null}
                                    aria-controls={null}
                                    className="euiComboBox__input"
                                    data-test-subj="comboBoxSearchInput"
                                    id="index"
                                    injectStyles={true}
                                    inputRef={[Function]}
                                    minWidth={1}
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onFocus={[Function]}
                                    role="textbox"
                                    style={
                                      Object {
                                        "fontSize": 14,
                                      }
                                    }
                                    value=""
                                  >
                                    <div
                                      className="euiComboBox__input"
                                      style={
                                        Object {
                                          "display": "inline-block",
                                          "fontSize": 14,
                                        }
                                      }
                                    >
                                      <input
                                        aria-activedescendant={null}
                                        aria-controls={null}
                                        data-test-subj="comboBoxSearchInput"
                                        id="index"
                                        onBlur={[Function]}
                                        onChange={[Function]}
                                        onFocus={[Function]}
                                        role="textbox"
                                        style={
                                          Object {
                                            "boxSizing": "content-box",
                                            "width": "2px",
                                          }
                                        }
                                        value=""
                                      />
                                      <div
                                        style={
                                          Object {
                                            "height": 0,
                                            "left": 0,
                                            "overflow": "scroll",
                                            "position": "absolute",
                                            "top": 0,
                                            "visibility": "hidden",
                                            "whiteSpace": "pre",
                                          }
                                        }
                                      />
                                    </div>
                                  </AutosizeInput>
                                </div>
                                <EuiFormControlLayoutIcons
                                  icon={
                                    Object {
                                      "aria-label": "Open list of options",
                                      "data-test-subj": "comboBoxToggleListButton",
                                      "disabled": undefined,
                                      "onClick": [Function],
                                      "ref": [Function],
                                      "side": "right",
                                      "type": "arrowDown",
                                    }
                                  }
                                  isLoading={false}
                                >
                                  <div
                                    className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                                  >
                                    <EuiFormControlLayoutCustomIcon
                                      aria-label="Open list of options"
                                      data-test-subj="comboBoxToggleListButton"
                                      iconRef={[Function]}
                                      onClick={[Function]}
                                      type="arrowDown"
                                    >
                                      <button
                                        aria-label="Open list of options"
                                        className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                                        data-test-subj="comboBoxToggleListButton"
                                        onClick={[Function]}
                                        type="button"
                                      >
                                        <EuiIcon
                                          aria-hidden="true"
                                          className="euiFormControlLayoutCustomIcon__icon"
                                          type="arrowDown"
                                        >
                                          <arrowDown
                                            aria-hidden="true"
                                            className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                                            focusable="false"
                                            height="16"
                                            style={null}
                                            viewBox="0 0 16 16"
                                            width="16"
                                            xmlns="http://www.w3.org/2000/svg"
                                            xmlnsXlink="http://www.w3.org/1999/xlink"
                                          >
                                            <svg
                                              aria-hidden="true"
                                              className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                                              focusable="false"
                                              height="16"
                                              style={null}
                                              viewBox="0 0 16 16"
                                              width="16"
                                              xmlns="http://www.w3.org/2000/svg"
                                              xmlnsXlink="http://www.w3.org/1999/xlink"
                                            >
                                              <defs>
                                                <path
                                                  d="M13.069 5.157L8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0z"
                                                  id="arrow_down-a"
                                                />
                                              </defs>
                                              <use
                                                fillRule="nonzero"
                                                xlinkHref="#arrow_down-a"
                                              />
                                            </svg>
                                          </arrowDown>
                                        </EuiIcon>
                                      </button>
                                    </EuiFormControlLayoutCustomIcon>
                                  </div>
                                </EuiFormControlLayoutIcons>
                              </div>
                            </div>
                          </EuiFormControlLayout>
                        </EuiComboBoxInput>
                      </div>
                    </EuiComboBox>
                  </ComboBox>
                  <EuiFormHelpText
                    className="euiFormRow__text"
                    id="index-help"
                  >
                    <div
                      className="euiFormHelpText euiFormRow__text"
                      id="index-help"
                    >
                      You can use a * as a wildcard in your index pattern
                    </div>
                  </EuiFormHelpText>
                </div>
              </EuiFormRow>
            </FormikFormRow>
          </FieldInner>
        </C>
      </FormikInputWrapper>
    </FormikComboBox>
  </MonitorIndex>
</Formik>
`;
